bindings/python/htcondor/_utils/ansi.py

```diff
@@ -29,33 +30,87 @@
 IS_WINDOWS = os.name == "nt"
 IS_TTY = sys.stdout.isatty()
 
+ANSI_RESET = "\033[0m"
+ANSI_ESCAPE_RE = re.compile(r"\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])")
 
 class Color(str, enum.Enum):
     BLACK = "\033[30m"
     RED = "\033[31m"
-    BRIGHT_RED = "\033[31;1m"
     GREEN = "\033[32m"
-    BRIGHT_GREEN = "\033[32;1m"
     YELLOW = "\033[33m"
-    BRIGHT_YELLOW = "\033[33;1m"
     BLUE = "\033[34m"
-    BRIGHT_BLUE = "\033[34;1m"
     MAGENTA = "\033[35m"
-    BRIGHT_MAGENTA = "\033[35;1m"
     CYAN = "\033[36m"
-    BRIGHT_CYAN = "\033[36;1m"
     WHITE = "\033[37m"
-    BRIGHT_WHITE = "\033[37;1m"
-    RESET = "\033[0m"
+    DARK_GREY = "\033[90m"
+    BRIGHT_RED = "\033[91m"
+    BRIGHT_GREEN = "\033[92m"
+    BRIGHT_YELLOW = "\033[93m"
+    BRIGHT_BLUE = "\033[94m"
+    BRIGHT_MAGENTA = "\033[95m"
+    BRIGHT_CYAN = "\033[96m"
+    BRIGHT_WHITE = "\033[97m"
 
+class Style(str, enum.Enum):
+    BOLD = "\033[1m"
+    UNDERLINE = "\033[4m"
+    BLINK = "\033[5m"
+    # ITALIC = "\033[3m" Note: Not widely supported
 
-def can_color():
+class AnsiOptions():
+    def __init__(self, **kwargs):
+        self.bold = kwargs.get("bold", False)
+        self.blink = kwargs.get("blink", False)
+        self.underline = kwargs.get("underline", False)
+        self.style = kwargs.get("style")
+        self.color = kwargs.get("color")
+        self.color_id = kwargs.get("color_id")
+
+        if self.style is not None:
+            self.bold = True if self.style == Style.BOLD else self.bold
+            self.blink = True if self.style == Style.BLINK else self.blink
+            self.underline = True if self.style == Style.UNDERLINE else self.underline
+
+        if self.color is not None:
+            self.color = self.color[2:-1]
+
+        if self.color_id is not None:
+            self.color_id = sorted([0, self.color_id, 255])
```

Comment:
It's strongly implied by the rest of the code that you expect `self.color_id` to be a single integer; this could should be returning a list.  Perhaps you meant

```suggestion
            self.color_id = sorted([0, self.color_id, 255])[1]
```
?

Reply:
(This isn't a new bug, if it is one.)

Reply:
You are correct... `color_id` is supposed to be a single integer I missed the number selection. Good catch